File: backend/app/model/detection.py
from app.extensions import db
from .timestamps import TimestampedModel

class Detection(TimestampedModel):
    __tablename__ = "detections"

    detection_id = db.Column(db.Integer, primary_key=True)

    camera_id = db.Column(
        db.Integer,
        db.ForeignKey("cameras.camera_id", ondelete="CASCADE"),
        nullable=False
    )

    track_id = db.Column(
        db.Integer,
        db.ForeignKey("tracks.track_id", ondelete="CASCADE"),
        nullable=False
    )

    coordinates = db.Column(db.JSON, nullable=False)

File: Frontend/src/types/detection.ts
export interface DetectionCoordinates {
  x1: number;
  y1: number;
  x2: number;
  y2: number;
}

export interface Detection {
  detection_id: number;
  camera_id: number;
  track_id: number;
  coordinates: DetectionCoordinates;
  created_at: string;
  updated_at: string;
}
